applications/argocd/staging/applications/alert-hub-frontend.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: alert-hub-frontend
+  namespace: argocd
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  project: default
+  source:
+    repoURL: ghcr.io/ifrcgo/alert-hub-web-app
+    chart: ifrcgo-web-app-nginx-serve
+    targetRevision: 0.0.1-develop.d39f061
+    helm:
+      parameters:
+        - name: ingress.host
+          value: "alerthub-stage.ifrc.org"
+        - name: ingress.className
+          value: "nginx"
+        - name: env.APP_MAPBOX_ACCESS_TOKEN
+          value: "pk.ab.xy"
```

Comment:
We need additional variables.

```
APP_TITLE: "Alert hub"
APP_MAPBOX_ACCESS_TOKEN: pk.eyJ1IjoiZ28taWZyYyIsImEiOiJjams3b2ZhZWswMGFvM3hxeHp2ZHFhOTRrIn0._pqO9OQ2iNeDGrpopJNjpg
APP_GOOGLE_ANALYTICS_ID: "G-PZFCBMTDPV"

# This will be different per environment
# - PROD
APP_ENVIRONMENT: production
APP_GRAPHQL_API_ENDPOINT: https://alerthub-api.ifrc.org/graphql/
APP_HCAPTCHA_SITEKEY: eda042c2-b734-4bb0-80f8-e34c1752609c

# - Staging
APP_ENVIRONMENT: staging
APP_GRAPHQL_API_ENDPOINT: https://alerthub-stage-api.ifrc.org/graphql/
APP_HCAPTCHA_SITEKEY: 35f8a154-7f0c-46fa-ae3e-9000fbc86c8e
```

Reply:
@thenav56, wouldn't it be better to supply these through appropriate values files and leave the values defined here to be only absolutely necessary overrides? I'm saying this because I think having application config in too many places could be a potential source of error.

Reply:
Also, the values file being referred to in the ArgoCD Application resource here are those in the respective backend/frontend repo. So the values-staging.yaml file should be in the backend repo.

Reply:
@thenav56 , staging parameters have now been included for the frontend. Have another look.